fastify.js

```diff
@@ -173,7 +173,14 @@ function build (options) {
   fastify.setSchemaCompiler(buildSchemaCompiler())
 
   // plugin
-  fastify.register = fastify.use
+  fastify.avvioUse = fastify.use
+  fastify.register = function (plugin, options) {
+    fastify.avvioUse(plugin, options)
+    if (plugin.name.length === 0) {
+      log.warn('Warning: should not use anonymous plugins. Provide a function name on declaration.')
+    }
```

Comment:
I don’t think emitting a warn is feasible. We should have default strategy (maybe the filename? or the closest name in package.json) in case there is no name.

Reply:
How would this method have access to the filename? Even if the user uses:
```javascript
// plugin.js
const fp = require('fastify-plugin')

module.exports = fp((instance, opts, next) => {
  instance.decorate('mult', (a, b) => a * b)
  next()
})
```

```javascript
// app.js
fastify.register(require('plugin.js'))
```

The `register` method will still receive an anonymous function.